ccips/ccip-009/ccip-009-citycoins-vrf-v2.md

```diff
@@ -0,0 +1,214 @@
+
+```
++----------------------+----------+------------+------------+-----------+
+|                      | Consumed | Limit      | Percent    | Diff (v1) |
++----------------------+----------+------------+------------+-----------+
+| Runtime              | 59665    | 5000000000 | 0.001%     | -0.007%   |
++----------------------+----------+------------+------------+-----------+
+| Read count           | 6        | 7750       | 0.077%     | +0.025%   |
++----------------------+----------+------------+------------+-----------+
+| Read length (bytes)  | 2221     | 100000000  | 0.002%     | -0.001%   |
++----------------------+----------+------------+------------+-----------+
+| Write count          | 1        | 7750       | 0.013%     | +0.013%   |
++----------------------+----------+------------+------------+-----------+
+| Write length (bytes) | 33       | 15000000   | 0.000%     | +0.000%   |
++----------------------+----------+------------+------------+-----------+
+```
+
+v2 VRF `get-save-rnd`, on subsequent calls after value is saved
+
+```
++----------------------+----------+------------+------------+-----------+
+|                      | Consumed | Limit      | Percent    | Diff (v1) |
++----------------------+----------+------------+------------+-----------+
+| Runtime              | 4782     | 5000000000 | 0.000%     | -0.008%   |
++----------------------+----------+------------+------------+-----------+
+| Read count           | 4        | 7750       | 0.052%     | same      |
++----------------------+----------+------------+------------+-----------+
+| Read length (bytes)  | 2220     | 100000000  | 0.002%     | -0.001%   |
++----------------------+----------+------------+------------+-----------+
+| Write count          | 0        | 7750       | 0.000%     | same      |
++----------------------+----------+------------+------------+-----------+
+| Write length (bytes) | 0        | 15000000   | 0.000%     | same      |
++----------------------+----------+------------+------------+-----------+
+```
+
+#### Claim Mining Reward
+
+v1 `claim-mining-reward`
+
+```
++----------------------+----------+------------+------------+
+|                      | Consumed | Limit      | Percent    |
++----------------------+----------+------------+------------+
+| Runtime              | 527214   | 5000000000 | 0.011%     |
++----------------------+----------+------------+------------+
+| Read count           | 32       | 7750       | 0.413%     |
++----------------------+----------+------------+------------+
+| Read length (bytes)  | 58176    | 100000000  | 0.058%     |
++----------------------+----------+------------+------------+
+| Write count          | 5        | 7750       | 0.065%     |
++----------------------+----------+------------+------------+
+| Write length (bytes) | 459      | 15000000   | 0.003%     |
++----------------------+----------+------------+------------+
+```
+
+v2 `claim-mining-reward` on first call before value is saved
```

Comment:
```suggestion
v2 `claim-mining-reward` when `rnd` value has not been generated and saved by anyone before
```
`claim-mining-reward` can be successfully called only once per block, therefore we should avoid "first call" term.

Reply:
Fixed in https://github.com/citycoins/governance/pull/5/commits/f4f44b8e6e3cc5cb7ef68d215c29c2cf1676f06f, took a slightly different approach but agree that could've been confusing. Thanks!